libminifi/include/core/state/Value.h

```diff
@@ -380,9 +411,59 @@ class Int64Value : public Value {
     return false;
   }
 
+  virtual bool getValue(double& /*ref*/) {
+    return false;
+  }
+
   int64_t value;
 };
 
+class DoubleValue : public Value {
+ public:
+  explicit DoubleValue(double value)
+      : Value(std::to_string(value)),
+      value(value) {
+    setTypeId<double>();
+  }
+  explicit DoubleValue(const std::string &strvalue)
+      : Value(strvalue) {
+    utils::internal::ValueParser(strvalue).parse(value).parseEnd();
+    setTypeId<double>();
+  }
+
+  double getValue() {
+    return value;
+  }
+
+ protected:
+  virtual bool getValue(int&) {
+    return false;
+  }
+
+  virtual bool getValue(uint32_t&) {
+    return false;
+  }
+
+  virtual bool getValue(int64_t&) {
+    return false;
```

Comment:
why disable all castings? others like `IntValue` check the range (quite badly) and cast if possible, should we provide a helper like this, and thread all these `Value::getValue(*)` methods body through it? 
```
template<T, U>
bool utils::cast_if_in_range(T in, U& out) {
  U result = static_cast<U>(in);
  if (static_cast<T>(result) != in) {
    return false;
  }
  out = result;
  return true;
   
}
```

Reply:
Thanks, I've added this (extended to work with signed-unsigned casts) function and changed to use it where it makes sense in [d28d399](https://github.com/apache/nifi-minifi-cpp/pull/1032/commits/d28d3994fe633513158fb739edc49f69023b8a39)
Also added a bunch more unit tests to verify the behaviour.